### ISSUE TITLE
Bug: Active Merchant CIM does not correctly receive bank_accounts in options 

### ISSUE THREAD

**Issue by eddkerr** [activemerchant/active_merchant#2353]
Cards seem to work fine, but when submitting a bank account to create_new_customer_payment_profile it raises the following error: 
"undefined method `[]' for #<ActiveMerchant::Billing::Check:0x007fba826f4b60>"

    options = { account_number: '**********',
                    routing_number: '**********',
                    first_name: user.first_name,
                    last_name: user.last_name,
                    account_type: 'checking' }

        bank_account = ActiveMerchant::Billing::Check.new(options)

        response = AUTHORIZE_CIM.create_customer_payment_profile(customer_profile_id: '**********', payment_profile: { payment: { bank_account: ba } })

If you submit the options directly as follows

    response = AUTHORIZE_CIM.create_customer_payment_profile(customer_profile_id: '**********', payment_profile: { payment: { bank_account: options } })

It raises the error "Invalid Bank Account Type: checking"

**Comment by filipebarcos** [comment]
@eddkerr Seems like you're referencing some code from your codebase and we're lacking some context here. Is this still a issue for you, if so, can you provide more info? Or can we close this?

**Comment by eddkerr** [comment]
Hey Filipe,

I was able to fix the problem. I have later discovered that active merchant
might not be supporting what it is that I am attempting to do. The issue
was that the add_bank_account method was looking for a hash, when it needed
to be looking for an object! It now works the same way that add_credit_card
does, and it actually works. I hope this clarifies things, but the portions
of the code that I touched seemed like they could never work.
